packages/web-components/src/components/rux-calendar/rux-calendar.tsx

```diff
@@ -0,0 +1,547 @@
+    handleMonthWatch() {
+        console.log('heard month change')
+        this._updateDate(this._year, this._month)
+    }
+
+    @Watch('_year')
+    handleYearWatch() {
+        console.log('heard year change in WATCH')
+        // this._handleYears(this._maxDate, this._minDate)
+        this._updateDate(this._year, this._month)
+    }
+
+    //? Might need later when/if we tackle dynamic min/max changes
+    // @Watch('min')
+    // @Watch('max')
+    // handleMinMaxChange() {
+    //     console.log('heard min or max change')
+    // }
+
+    private _currentDate: Date = utcToZonedTime(new Date(Date.now()), 'UTC')
+    // private _year: number = this._date.getFullYear()
+    private _currentDay: number = this._currentDate.getDate()
+    private _daysInMonth: number = getDaysInMonth(this._date)
+    private _daysInMonthArr: Array<number> = []
+    private _prevMonth: number = this._month - 1 < 1 ? 12 : this._month - 1
+    private _nextMonth: number = this._month + 1 >= 13 ? 1 : this._month + 1
+    private _prevDaysToShow: { [key: string]: any } = {}
+    private _nextDaysToShow: Array<Number> = this._findNextDaysToShow()
+    private _allYearsArr: Array<Number> = []
+
+    //Default the max/min Dates to be 10 years in either direciton.
+    private _maxDate: Date = this.max
+        ? new Date(this.max)
+        : new Date(`${this._date.getFullYear() + 11}-01-01`)
+    private _minDate: Date = this.min
+        ? new Date(this.min)
+        : new Date(`${this._date.getFullYear() - 9}-01-01`)
+    private _maxYearArr: Array<Number> = []
+    private _minYearArr: Array<Number> = []
+
+    connectedCallback() {
+        console.log('CC Fire')
+
+        this._fillDaysInMonthArr()
+        this._nextDaysToShow = this._findNextDaysToShow()
+        this._handleMonthChange = this._handleMonthChange.bind(this)
+        this._handleYearChange = this._handleYearChange.bind(this)
+
+        if (this.dateIn) {
+            console.log('true')
+            console.log(this._maxDate, 'max date')
+            console.log(this._minDate, 'min date')
+            this._setStateWithDateIn()
+        }
+        this._handleYears(this._maxDate, this._minDate)
+    }
+    componentDidLoad() {
+        // attach event listeners to back and forward month arrows
+        this.el
+            .shadowRoot!.querySelector('#backward-month')
+            ?.addEventListener('click', () => {
+                this._handleBackwardArrow()
+            })
+        this.el
+            .shadowRoot!.querySelector('#forward-month')
+            ?.addEventListener('click', () => {
+                this._handleForwardArrow()
+            })
+    }
+    //? Want to remove the selected prop from rux-day if changing month/year. Is this a good way to do it?
+    //? what kind of side effects can this have? What other times will componentWillUpdate fire?
+    //? Might become an issue when we use the month/year picker to change month/year?
+    componentWillUpdate() {
```

Comment:
Here is my suggestion: `componentWillUpdate() {
        const days = this.el.shadowRoot!.querySelectorAll('rux-day')
        days.forEach((day: HTMLRuxDayElement) => {
          if (day.selected) day.selected = false
        })
    }`